homeassistant/components/weatherflow/sensor.py

```diff
@@ -0,0 +1,417 @@
+from pyweatherflowudp.const import EVENT_RAPID_WIND
+from pyweatherflowudp.device import (
+    EVENT_OBSERVATION,
+    EVENT_STATUS_UPDATE,
+    WeatherFlowDevice,
+)
+
+from homeassistant.components.sensor import (
+    DOMAIN as SENSOR_DOMAIN,
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    DEGREE,
+    LIGHT_LUX,
+    PERCENTAGE,
+    SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+    UV_INDEX,
+    EntityCategory,
+    UnitOfElectricPotential,
+    UnitOfIrradiance,
+    UnitOfLength,
+    UnitOfPrecipitationDepth,
+    UnitOfPressure,
+    UnitOfSpeed,
+    UnitOfTemperature,
+    UnitOfVolumetricFlux,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.util.unit_system import METRIC_SYSTEM
+
+from .const import DOMAIN, LOGGER
+
+
+@dataclass
+class WeatherFlowSensorEntityDescription(SensorEntityDescription):
+    """Describes a WeatherFlow sensor entity description."""
+
+    event_subscriptions: list[str] = field(default_factory=lambda: [EVENT_OBSERVATION])
+    imperial_suggested_unit: None | str = None
+
+    backing_library_attribute_fn: Callable[[], str] = lambda: ""
+
+
+@dataclass
+class AirDensityWeatherFlowSensorEntityDescription(WeatherFlowSensorEntityDescription):
+    """Custom class to handle the conversion between backing lib and Home Assistant Compatible VOC sensor."""
+
+    imperial_unit_of_measurement: str | None = None
+
+
+@dataclass
+class WeatherFlowWindSensorEntityDescription(WeatherFlowSensorEntityDescription):
+    """Describes a WeatherFlow wind sensor entity description."""
+
+    def __post_init__(self) -> None:
+        """Post initialisation processing."""
+        self.icon = "mdi:weather-windy"
+        self.native_unit_of_measurement = UnitOfSpeed.KILOMETERS_PER_HOUR
+        self.state_class = SensorStateClass.MEASUREMENT
+        self.suggested_display_precision = 2
+
+
+CUSTOM_SENSORS: tuple[AirDensityWeatherFlowSensorEntityDescription, ...] = (
+    AirDensityWeatherFlowSensorEntityDescription(
+        key="air_density",
+        translation_key="air_density",
+        native_unit_of_measurement="µg/m³",
+        device_class=SensorDeviceClass.VOLATILE_ORGANIC_COMPOUNDS,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=5,
+    ),
+)
+
+SENSORS: tuple[WeatherFlowSensorEntityDescription, ...] = (
+    WeatherFlowSensorEntityDescription(
+        key="air_temperature",
+        translation_key="temperature",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
```

Comment:
Every sensor that uses something like Temperature, Humidity, Windspeed, etc as device class and the name is equal, you can remove the translations key or name, since it will use the name of the device classes.

Reply:
Only when the name matches the device class. sooo

"Temperature" -> name and translation key can be omitted as they are the same
"Feels like" -> Although it has the device class temperature, it isn't called Temperature so we keep it that way